src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.cs

```diff
@@ -155,44 +155,45 @@
+    ("_ImmOpTestTemplate.template",                        "ImmOpTest.template",                           new Dictionary<string, string> { ["TemplateName"] = "Imm",        ["TemplateValidationLogic"] = SimpleOpTest_ValidationLogic }),
+    ("_ImmBinaryOpTestTemplate.template",                  "ImmBinOpTest.template",                        new Dictionary<string, string> { ["TemplateName"] = "Imm",        ["TemplateValidationLogic"] = SimpleOpTest_ValidationLogic }),
+    ("_ImmBinaryOpTestTemplate.template",                  "VecImmBinOpTest.template",                     new Dictionary<string, string> { ["TemplateName"] = "Imm",        ["TemplateValidationLogic"] = SimpleVecOpTest_ValidationLogic }),
+    ("_BinaryOpTestTemplate.template",                     "SimpleBinOpTest.template",                     new Dictionary<string, string> { ["TemplateName"] = "Simple",     ["TemplateValidationLogic"] = SimpleOpTest_ValidationLogic }),
+    ("_TernaryOpTestTemplate.template",                    "VecTernOpTest.template",                       new Dictionary<string, string> { ["TemplateName"] = "Simple",     ["TemplateValidationLogic"] = SimpleVecOpTest_ValidationLogic }),
+    ("_ImmTernaryOpTestTemplate.template",                 "VecImmTernOpTest.template",                    new Dictionary<string, string> { ["TemplateName"] = "Simple",     ["TemplateValidationLogic"] = SimpleVecOpTest_ValidationLogic }),
+    ("_ImmTernaryOpTestTemplate.template",                 "SimpleImmTernOpTest.template",                 new Dictionary<string, string> { ["TemplateName"] = "Simple",     ["TemplateValidationLogic"] = SimpleOpTest_ValidationLogic }),
+    ("_UnaryOpTestTemplate.template",                      "SimpleUnOpTest.template",                      new Dictionary<string, string> { ["TemplateName"] = "Simple",     ["TemplateValidationLogic"] = SimpleOpTest_ValidationLogic }),
+    ("_UnaryOpTestTemplate.template",                      "SimpleVecOpTest.template",                     new Dictionary<string, string> { ["TemplateName"] = "Simple",     ["TemplateValidationLogic"] = SimpleVecOpTest_ValidationLogic }),
+    ("_BinaryOpTestTemplate.template",                     "VecPairBinOpTest.template",                    new Dictionary<string, string> { ["TemplateName"] = "Simple",     ["TemplateValidationLogic"] = VecPairBinOpTest_ValidationLogic }),
+    ("_BinaryOp_SveTestTemplate.template",                 "SveVecPairBinOpTest.template",                 new Dictionary<string, string> { ["TemplateName"] = "Simple",     ["TemplateValidationLogic"] = VecPairBinOpTest_ValidationLogic }),
+    ("_BinaryOpDifferentOpTypes_SveTestTemplate.template", "SveVecPairBinOpDifferentOpTypesTest.template", new Dictionary<string, string> { ["TemplateName"] = "Simple",     ["TemplateValidationLogic"] = VecPairBinOpTest_ValidationLogic }),
```

Comment:
Any reason why we cannot use `SveVecBinOpTest*.template`?

Reply:
The `SveVecBinOpTest` is built for when the 2 arguments are of the same type. I made `SveVecPairBinOpDifferentOpTypesTest` to be the flexible variation.

Reply:
Is it possible to tweak `Op1BaseType` and `Op2BaseType` and reuse `SveVecBinOpTest`? You might need some casting, but wondering if it is easy to do?